addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -219,49 +220,57 @@ func (r *MirrorPeerReconciler) enableCSIAddons(ctx context.Context, namespace st
 }
 
-func (r *MirrorPeerReconciler) createVolumeReplicationClass(ctx context.Context, mp *multiclusterv1alpha1.MirrorPeer) error {
+func (r *MirrorPeerReconciler) createVolumeReplicationClass(ctx context.Context, mp *multiclusterv1alpha1.MirrorPeer) []error {
 	scr, err := common.GetCurrentStorageClusterRef(mp, r.SpokeClusterName)
+	var errs []error
 	if err != nil {
 		klog.Error(err, "Failed to get current storage cluster ref")
-		return err
-	}
-	params := make(map[string]string)
-	params[MirroringModeKey] = string(mp.Spec.Mode)
-	params[SchedulingIntervalKey] = mp.Spec.SchedulingInterval
-	params[ReplicationSecretNameKey] = mp.Spec.ReplicationSecretName
-	params[ReplicationSecretNamespaceKey] = scr.Namespace
-	vrc := &replicationv1alpha1.VolumeReplicationClass{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "odf-rbd-volumereplicationclass",
-		},
-		Spec: replicationv1alpha1.VolumeReplicationClassSpec{
-			Provisioner: fmt.Sprintf(ProvisionerTemplate, scr.Namespace),
-			Parameters:  params,
-		},
+		errs = append(errs, err)
 	}
-	found := &replicationv1alpha1.VolumeReplicationClass{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "odf-rbd-volumereplicationclass",
-		},
+	if len(errs) > 0 {
+		return errs
 	}
-	err = r.SpokeClient.Get(ctx, types.NamespacedName{
-		Name: found.Name,
-	}, found)
+	for _, interval := range mp.Spec.SchedulingIntervals {
+		params := make(map[string]string)
+		params[MirroringModeKey] = DefaultMirroringMode
+		params[SchedulingIntervalKey] = interval
+		params[ReplicationSecretNameKey] = ReplicationSecretName
+		params[ReplicationSecretNamespaceKey] = scr.Namespace
+		vrcName := fmt.Sprintf(VolumeReplicationClassNameTemplate, common.FnvHash(interval))
+		found := &replicationv1alpha1.VolumeReplicationClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: vrcName,
+			},
+		}
+		err = r.SpokeClient.Get(ctx, types.NamespacedName{
+			Name: found.Name,
+		}, found)
 
-	if err != nil && !errors.IsNotFound(err) {
-		return err
-	}
-	op, err := controllerutil.CreateOrUpdate(ctx, r.SpokeClient, found, func() error {
-		found.Spec = vrc.Spec
-		return nil
-	})
-	if err != nil {
-		klog.Error(err, "Failed to create/update volume replication class")
-		return err
-	} else {
-		klog.Info("Volume replication class successfully ", op)
+		switch {
+		case err == nil:
+			klog.Infof("VolumeReplicationClass already exists: %s", vrcName)
+			continue
+		case !errors.IsNotFound(err):
+			klog.Error(err, "Failed to get VolumeReplicationClass: %s", vrcName)
+			errs = append(errs, err)
+			continue
+		}
+		vrc := replicationv1alpha1.VolumeReplicationClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: vrcName,
+			},
+			Spec: replicationv1alpha1.VolumeReplicationClassSpec{
+				Parameters: params,
+			},
+		}
+		err = r.SpokeClient.Create(ctx, &vrc)
+		if err != nil {
+			klog.Error(err, "Failed to create VolumeReplicationClass: %s", vrcName)
+			errs = append(errs, err)
+		}
+		klog.Infof("VolumeReplicationClass created: %s", vrcName)
```

Comment:
this will log both failure and success as there is no break or return statement.

Reply:
fixed with continue